internal/secrets/encryptor_test.go

```
@@ -0,0 +1,285 @@
+			t.Fatal("unable to decrypt and get the original bytes")
+		}
+	})
+}
+
+func TestNoOpEncryptor(t *testing.T) {
+	e := noOpEncryptor{}
+
+	encrypted, err := e.EncryptBytes(messageToEncrypt)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(encrypted, messageToEncrypt) {
+		t.Fatal("encrypted bytes is not same as the original")
+	}
+}
+
+func TestBadKeysFailToDecrypt(t *testing.T) {
+	key, err := generateRandomAESKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	e := newEncryptor(key, nil)
+
+	encrypted, err := e.EncryptBytes(messageToEncrypt)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	decrypted, err := e.DecryptBytes(encrypted)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	notTheSameKey, err := generateRandomAESKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+	e = newEncryptor(notTheSameKey, nil)
+
+	decryptedAgain, err := e.DecryptBytes(encrypted)
+	if err == nil {
+		t.Fatal("Should not have been able to Decrypt string with an invalid secret key.")
+	}
+
+	if bytes.Equal(decrypted, decryptedAgain) {
+		t.Fatal("Should not have been able to Decrypt string with an invalid secret key.")
+	}
+}
+
+// Test that different strings EncryptBytes to different outputs
+func TestDifferentOutputs(t *testing.T) {
+	key, _ := generateRandomAESKey()
+	e := newEncryptor(key, nil)
+	messages := []string{
+		"This may or may",
+		"This is not the same as that",
+		"The end of that",
+		"Plants and animals",
+		"Snow, igloos, sunshine, unicords",
+	}
+
+	var crypts []string
+	for _, m := range messages {
+		encrypted, _ := e.EncryptBytes([]byte(m))
+		crypts = append(crypts, string(encrypted))
+	}
+
+	for _, c := range crypts {
+		if !isInSliceOnce(c, crypts) {
+			t.Fatalf("Duplicate encryption string: %v.", c)
+		}
+	}
+}
+
+func isInSliceOnce(item string, slice []string) bool {
+	found := 0
+	for _, s := range slice {
+		if item == s {
+			found++
+		}
+	}
+
+	return found == 1
+}
+
+func TestSampleNoRepeats(t *testing.T) {
+	key, _ := generateRandomAESKey()
```

Check the error or add a code comment to explain why it is OK to ignore the error here in particular.